src/services/lnd/index.ts

```diff
@@ -462,6 +462,27 @@ const lookupPaymentByPubkeyAndHash = async ({
   }
 }
 
+// A rough description of the error type we get back from the
+// 'lightning' library can be described as:
+//
+// [
+//   0: <Error Classification Code Number>
+//   1: <Error Type String>
+//   2: {
+//     err?: <Error Code Details Object>
+//     failures?: [
+//         0: <Error Code Number>
+//         1: <Error Code Message String>
+//         2: {
+//           err?: <Error Code Details Object>
+//         }
+//     ][]
+//   }
+// ]
+//
+// where '<Error Code Details Object>' is an Error object with
+// the usual 'message', 'stack' etc. properties and additional
+// properties: 'code', 'details', 'metadata'.
 const parseLndErrorDetails = (err) =>
   err[2]?.err?.details || err[2]?.failures?.[0]?.[2]?.err?.details || err[1]
```

Comment:
shouldn't it be `err[2]?.failures?.[2]?.err?.details` then? (dropping a `[0]`)

Reply:
`failures` is an array of failure objects (where was object has the 0/1/2 indices). I tried to capture this with the `[]` at the very end of that failures object.

Reply:
yup ok

Reply:
[Refactored](https://github.com/GaloyMoney/galoy/compare/8b5e80ebeeec87b4f9b1d42e5c547739a78b1c8a..2c07f214fb63bf88559d4b9b3145fc596449365e) to hopefully make that clearer